src/components/form/form-password-input/form-password-input.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react';
+import { TextInput } from 'react-native-paper';
+
+import { FormInput } from 'src/components';
+import styles from '../styles';
+
+type FormInputProps = React.ComponentPropsWithoutRef<typeof FormInput>;
+
+type FormPasswordInputProps = Omit<FormInputProps, 'right'>;
```

Comment:
I don't think we need to extract `right` here. We might want to override it sometime in future.